src/python/pants/engine/legacy/graph.py

```diff
@@ -463,20 +458,16 @@ def validate(self, *, pants_bin_name: str) -> None:
 
 
 @rule
-async def find_owners(
-  build_configuration: BuildConfiguration,
-  address_mapper: AddressMapper,
-  owners_request: OwnersRequest
-) -> BuildFileAddresses:
+async def find_owners(owners_request: OwnersRequest) -> BuildFileAddresses:
   sources_set = OrderedSet(owners_request.sources)
-  dirs_set = OrderedSet(dirname(source) for source in sources_set)
+  dirs_set = OrderedSet(os.path.dirname(source) for source in sources_set)
 
   # Walk up the buildroot looking for targets that would conceivably claim changed sources.
   candidate_specs = tuple(AscendantAddresses(directory=d) for d in dirs_set)
   candidate_targets = await Get[HydratedTargets](AddressSpecs(candidate_specs))
 
   # Match the source globs against the expanded candidate targets.
-  def owns_any_source(legacy_target):
+  def owns_any_source(legacy_target: HydratedTarget) -> bool:
```

Comment:
The whole purpose of this PR was to simplify this rule - everything else is either a necessary change to make this simplification possible or is a refactor.